lib/zipkin-tracer/application.rb

```diff
@@ -11,19 +11,19 @@ def self.routable_request?(path_info, http_method)
       false
     end
 
-    def self.get_route(path_info, http_method)
+    def self.get_route(env)
       return nil unless defined?(Rails)
-      req = Rack::Request.new("PATH_INFO" => path_info, "REQUEST_METHOD" => http_method)
+      req = Rack::Request.new(env)
       # Returns a string like /some/path/:id
-      Rails.application.routes.router.recognize(req) { |route|
+      Rails.application.routes.router.recognize(req) do |route|
         return route.path.spec.to_s
-      }
+      end
     rescue
       nil
     end
 
     def self.logger
-      if defined?(Rails) # If we happen to be inside a Rails app, use its logger
+      if defined?(Rails.logger) # If we happen to be inside a Rails app, use its logger
```

Comment:
1) Don't you need to keep the initial `Rails` check and _then_ check for the logger, e.g. 
```
if  defined?(Rails) &&  defined?(Rails.logger)
```

2) In what circumstances can Rails not have a logger defined?

Reply:
1. You don't have to keep the initial check. `defined?` won't raise an exception.

2. When ActiveRecord and ActionView are in use without Rails. (https://github.com/rails/rails-html-sanitizer/issues/26)

Reply:
thanks! 👍